coderadar-rest/src/main/java/io/reflectoring/coderadar/rest/query/GetCriticalFilesController.java

```diff
@@ -23,12 +26,52 @@ public GetCriticalFilesController(GetCriticalFilesUseCase getCriticalFilesUseCas
     this.getCriticalFilesUseCase = getCriticalFilesUseCase;
   }
 
-  @GetMapping(
+  @RequestMapping(
+      method = {RequestMethod.POST, RequestMethod.GET},
       path = "/projects/{projectId}/files/critical",
       produces = MediaType.APPLICATION_JSON_VALUE)
-  public ResponseEntity<List<ContributorsForFile>> getCriticalFiles(
-      @PathVariable Long projectId, @RequestBody @Validated GetCriticalFilesCommand command) {
+  public ResponseEntity<List<ContributorsForFile>> getFilesWithContributors(
+      @PathVariable Long projectId,
+      @RequestBody @Validated GetFilesWithContributorsCommand command) {
     return new ResponseEntity<>(
-        getCriticalFilesUseCase.getCriticalFiles(projectId, command), HttpStatus.OK);
+        getCriticalFilesUseCase.getFilesWithContributors(projectId, command), HttpStatus.OK);
+  }
+
+  @RequestMapping(
+      method = {RequestMethod.POST, RequestMethod.GET},
+      path = "/projects/{projectId}/files/modification/frequency")
+  public ResponseEntity<List<FileAndCommitsForTimePeriodResponse>> getFrequentlyChangedFiles(
+      @PathVariable long projectId,
+      @RequestBody @Validated GetFrequentlyChangedFilesCommand command) {
+    return new ResponseEntity<>(
+        mapResponse(getCriticalFilesUseCase.getFrequentlyChangedFiles(projectId, command)),
+        HttpStatus.OK);
+  }
+
+  private List<FileAndCommitsForTimePeriodResponse> mapResponse(
+      List<FileAndCommitsForTimePeriod> files) {
+    List<FileAndCommitsForTimePeriodResponse> result = new ArrayList<>(files.size());
+    for (FileAndCommitsForTimePeriod f : files) {
+      FileAndCommitsForTimePeriodResponse resultItem = new FileAndCommitsForTimePeriodResponse();
+      resultItem.setPath(f.getPath());
+      resultItem.setCommits(mapCommits(f.getCommits()));
+      result.add(resultItem);
+    }
+    return result;
+  }
+
+  private List<GetCommitResponse> mapCommits(List<Commit> commits) {
```

Comment:
You now have the same method in different classes. Maybe extract it to a `GetCommitResponseMapper` or something similar.

Reply:
Done